src/main.py

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env python3
+"""Play rock paper scissors on Twitter"""
+
+import os
+import enum
+import logging
+
+import tweepy  # type: ignore
+
+def configure_logger(module_name: str) -> logging.Logger:
+    """Configure the logger"""
+    logger = logging.getLogger(module_name)
+    formatter = logging.Formatter(fmt="%(levelname)s: %(message)s")
+    handler = logging.StreamHandler()
+    handler.setFormatter(formatter)
+    logger.addHandler(handler)
+    logger.setLevel(logging.INFO)
+    return logger
+
+LOG = configure_logger(__name__)
+
+class Weapon(enum.IntEnum):
+    """Enum of weapons"""
+    ROCK = 0
+    PAPER = 1
+    SCISSORS = 2
+
+    @property
+    def beaten_by(self) -> "Weapon":
+        """Creates a new weapon that will beat itself"""
+        return self.__class__((self.value + 1) % 3)
+
+    @property
+    def beats(self) -> "Weapon":
+        """Creates a new weapon that it will beat"""
+        return self.__class__((self.value - 1) % 3)
+
+class RockPaperScissors:
+    """Wrapper for the Twitter interface"""
+
+    MAX_COUNT = 1
+
+    def __init__(self) -> None:
+        auth = tweepy.OAuthHandler(os.environ["API_KEY"],
+                                   os.environ["API_SECRET"])
+        auth.set_access_token(os.environ["ACCESS_TOKEN"],
+                              os.environ["ACCESS_TOKEN_SECRET"])
+        self.api = tweepy.API(auth, wait_on_rate_limit=True,
+                              wait_on_rate_limit_notify=True)
+        self.me = self.api.me()  # pylint: disable=invalid-name
+        self.timeline = self.api.user_timeline(count=self.MAX_COUNT)
+
+    def _filter(self, tweet: tweepy.models.Status) -> bool:
+        if tweet.in_reply_to_user_id != self.me.id:
+            return False
+        if tweet.id in set(tweet.in_reply_to_status_id
+                           for tweet in self.timeline):
+            return False
+        weapons_in_tweet = set(weapon for weapon in Weapon
+                               if weapon.name in tweet.text.upper())
+        return len(weapons_in_tweet) == 1
+
+    @staticmethod
+    def _compose(tweet: tweepy.models.Status) -> dict:
+        weapon = set(weapon for weapon in Weapon
+                     if weapon.name in tweet.text.upper()).pop()
```

Comment:
No, it shouldn't respond to that text at all because the `_filter` will catch tweets that contain not exactly one weapon. It'll also title case it, so will respond with "Paper" rather than "PAPER".

However, I think you're possibly right that we should be more restrictive. Like, do we want to respond to things like "rocking"? When I wrote this I thought that we probably do.